server/gateway/src/controllers/loader.ts

```diff
@@ -56,7 +57,12 @@ export async function initialize(
         async () => Promise.resolve(resolved.tokens.storageToken),
         async () => Promise.resolve(resolved.tokens.socketToken)));
 
-    const tokenProvider = new GatewayTokenProvider(document.location.origin, resolved.url, hostToken, accessToken);
+    const tokenProvider = new GatewayTokenProvider(
+        document.location.origin,
+        resolved.url,
+        hostToken,
+        isSpoTenant(config.tenantId) ? resolved.tokens.storageToken : accessToken,
```

Comment:
Check in routes/loader.ts and send a flag in the controller to indicate whether its a spo tenant.

Reply:
Also you should do a if else check here as well. When its a SPO tenant, all the routerlicious driver code can be skipped

Reply:
So basically do the same thing you do in routes/loader.ts. Use if else to decide which driver you want.

Reply:
Loader routes now pass an isSpoTenantPath boolean to indicate which service factory to use in the controllers